adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,35 @@
 #include "impl.h"
+#include <algorithm>
+#include <stack>
 #include <vector>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source)
+{
+    std::stack<char> stack;
+    for (const auto& bracket : source) {
+        auto it = std::find_if(brackets.begin(), brackets.end(), [&](std::pair<char, char> const p) {
+            return p.first == bracket || p.second == bracket;
+        });
+        if (it == brackets.end()) {
+            continue;
+        }
+        if (it->first == bracket) {
+            stack.push(bracket);
+        } else {
+            if (stack.empty()) {
+                return false;
+            }
+            if (stack.top() != it->first) {
```

Comment:
please combine this condition with the above

Reply:
Done